labelSchema.js

```diff
@@ -83,6 +83,52 @@ function getFRACountryValue() {
   };
 }
 
+function isInNYC(record) {
+  const _region_a = getFirstProperty(['region_a'])(record);
+  const _country_a = getFirstProperty(['country_a'])(record);
+  const _locality_a = getFirstProperty(['locality_a'])(record);
+
+  return _country_a === 'USA' && _region_a === 'NY' && _locality_a === 'NYC';
+}
+
+function getUSABoroughValue(record) {
+  // In NYC, the borough is used as the locality name on address lines
+  // (except in Queens, see below), so don't return a borough at all
+  // in NYC if there's a locality value to return
+  if (isInNYC(record) && getUSALocalValue(record)) {
+    // Ignore the borough, it's handled in getUSALocalValue
+    return undefined;
+  }
+
+  return getFirstProperty(['borough'])(record);
+}
+
+function getUSALocalValue(record) {
+  const _default = getFirstProperty(['locality', 'localadmin', 'county'])(record);
+
+  // NYC is special for addresses
+  // - The borough is used for the locality in addresses
+  // - Except in Queens, where ideally the neighbourhood is
+  if (isInNYC(record)) {
+    const _borough = getFirstProperty(['borough'])(record);
+    const _neighbourhood = getFirstProperty(['neighbourhood'])(record);
+    // We still want to return "neighborhood, borough, region_a" when a user searches for a neighborhood
+    // otherwise it looks incomplete, so skip to returning the borough in that case
+    // Otherwise, in Queens only, use the neighborhood for the city in address labels
+    if ('neighbourhood' !== record.layer && 
+        _borough && 
+        _borough.startsWith('Queens') && 
```

Comment:
We don't have to solve this now, but I realized this won't work for most non-English languages, since translations will be applied before the labels are generated.

For example, when displaying results in Chinese: [/v1/search?lang=zh&text=44-45 21st St, Queens, NY 11101](https://pelias.github.io/compare/#/v1/search?lang=zh&text=44-45+21st+St%2C+Queens%2C+NY+11101&debug=1)

Maybe we could use WOF IDs instead? It's slightly more brittle but would work across all languages. Maybe it doesn't matter since NYC addresses probably don't actually have an official format in written languages other than English?